File: practice/programiz/make_a_simple_calculator_using_switch_case.c
#include <stdio.h>

/**
 * @brief this function add two numbers
 * @param first number: addition ke liye pehla number
 * @param second number: addition ke liye dusra number
 * @return the sum of the two numbers
 */
int addition(int first_number, int second_number)
{
    return first_number + second_number;
}

/***
 * @brief this function subtract two numbers
 * @param first number: subtraction ke liye pehla number
 * @param second number: subtraction ke liye dusra number
 * @return the difference of the two numbers
 */
int subtraction(int first_number, int second_number)
{
    return first_number - second_number;
}

/**
 * @brief this function multiply two numbers
 * @param first number: multiplication ke liye pehla number
 * @param second number: multiplication ke liye dusra number
 * @return the product of the two numbers
 */
int multiplication(int first_number, int second_number)
{
    return first_number * second_number;
}

/**
 * @brief this function divide two numbers
 * @param first number: division ke liye pehla number
 * @param second number: division ke liye dusra number
 * @return the quotient of the two numbers
 */
int division(int first_number, int second_number)
{
    return first_number / second_number;
}

/***
 * @brief this function find the remainder of two numbers
 * @param first number: remainder ke liye pehla number
 * @param second number: remainder ke liye dusra number
 * @return int: the remainder of the two numbers
 */
int calculate_remainder(int first_number, int second_number)
{
    return first_number % second_number;
}

/***
 * @brief this function performs the selected operation on two numbers
 * @param first number: the first number
 * @param second number: the second number
 * @param operation: the operation to be performed
 * @return the result of the operation
 */
int perform_operation(int first_number, int second_number, char operation)
{
    switch (operation)
    {
    case '+':
        return addition(first_number, second_number);
    case '-':
        return subtraction(first_number, second_number);
    case '*':
        return multiplication(first_number, second_number);
    case '/':
        if (second_number == 0)
        {
            printf("Error: division by zero is not allowed.\n");
            return 0;
        }
        return division(first_number, second_number);
    case '%':
        if (second_number == 0)
        {
            printf("Error: division by zero is not allowed.\n");
            return 0;
        }
        return calculate_remainder(first_number, second_number);
    default:
        printf("Invalid operation\n");
        return 0;
    }
}

int main(void)
{
    int first_number;
    int second_number;
    char operation;

    printf("Enter first number: ");
    if (scanf("%d", &first_number) != 1)
    {
        printf("Error: please enter a valid first number.\n");
        return 1;
    }

    printf("Enter second number: ");
    if (scanf("%d", &second_number) != 1)
    {
        printf("Error: please enter a valid second number.\n");
        return -1;
    }

    printf("Enter operation (+, -, *, /, %%): ");
    if (scanf(" %c", &operation) != 1)
    {
        printf("Error: please enter a valid operation.\n");
        return -1;
    }

    printf("Result: %d\n", perform_operation(first_number, second_number, operation));
    return 0;
}

File: practice/make_a_simple_calculator_using_switch_case.py
#@brief this function is used to add two numbers
#@param first_number: first number to add
#@param second_number: second number to add
#@return: sum of two numbers
def addition(first_number, second_number):
    return first_number + second_number
#@brief this function is used to subtract two numbers
#@param first_number: first number to subtract
#@param second_number: second number to subtract
#@return: difference of two numbers
def subtraction(first_number, second_number):
    return first_number - second_number

#@brief this function is used to multiply two numbers
#@param first_number: first number to multiply
#@param second_number: second number to multiply
#@return: product of two numbers
def multiplication(first_number, second_number):
    return first_number * second_number

#@brief this function is used to divide two numbers
#@param first_number: first number to divide
#@param second_number: second number to divide
#@return: quotient of two numbers

def division(first_number, second_number):
    return first_number / second_number

#@brief this function is used to find remainder of two numbers
#@param first_number: first number to find remainder    
#@param second_number: second number to find remainder
#@return: remainder of two numbers
def remainder(first_number, second_number):
    return first_number % second_number

print("Enter first number:")
first_number = int(input())
print("Enter second number:")
second_number = int(input())
print("Enter operator:/,+,-,*,/,%")
operator = input()
match operator:
    case "+":
        result = addition(first_number, second_number)
        print("result:" + str(result))
    case "-":
        result = subtraction(first_number, second_number)
        print("result:" + str(result))
    case "*":
        result = multiplication(first_number, second_number)
        print("result:" + str(result))
    case "/":
        if second_number == 0:
            print("division by zero is not allowed")
        else:
            result = division(first_number, second_number)
            print("result:" + str(result))
    case "%":
        if second_number == 0:
            print("division by zero is not allowed")
        else:
            result = remainder(first_number, second_number)
            print("result:" + str(result))
    case _:
        print("invalid operator")